src/util/metadata/side_metadata/mod.rs

```diff
@@ -0,0 +1,15 @@
+mod constants;
+mod helpers;
+#[cfg(target_pointer_width = "32")]
+mod helpers_32;
+
+mod global;
+mod sanity;
+mod side_metadata_tests;
+
+pub use constants::*;
```

Comment:
I assign the minimum visibility to all of the constants in the `constants.rs` where I define them, and the `pub use` here doesn't change the visibility scope.
It makes them accessible through `side_metadata::..` instead of `side_metadata::constants::..`, but only in their visibility scope.
So, for instance the bindings will only see global and local base addresses.

Let me know if you still think it is necessary to re-export one-by-one.

Reply:
In `util/mod.rs`, `side_metadata` was `pub(crate)` but `metadata` is `pub`. 
* You need to check `metadata`, as everything in it is public and there is a `pub use global::*`. 
* As `metadata::side_metadata` is public as well, you need to further check what is exposed in `metadata::side_metadata`.

My suggestion is to explicitly expose each item that is public. If there is anything that is clearly internal to mmtk-core, but needs to be exposed as public, then you would need to rethink about why it is needed in the binding.

Reply:
You may be right. Give me one sec to look through this.

Reply:
I took a look. Changes are still needed. There are constants that should not be public in `constants.rs`, e.g. `CHUNK_MASK`. I suggest you do a thorough check.

I suggest we do not expose internal modules. Instead we expose things at the level of `util::metadata`. For example,
* `util::metadata::MetadataSpec`
* `util::metadata::SideMetadataSpec` (if we should expose it)
* `util::metadata::InHeaderMetadataSpec`. 

We should hide our internal implementation details so if we do any internal refactoring, that does not affect our binding.

Reply:
If necessary we may have one more level in the path: `util::metadata::side_metadata`.

Reply:
I did a thorough check and reduced the visibility scope of some constants including `CHUNK_MASK`.

I believe having a separate `::side_metadata` level is helpful in that it encapsulates a meaningful part of the metadata, which is more internal to the core.